examples/cds-hooks/cds-hooks-launch.js

```diff
@@ -0,0 +1,194 @@
+        pem = fs.readFileSync(pemPath);
+      } catch (err) {
+        console.log(err);
+      }
+    } else if (typeof jku !== 'undefined') {
+      // Generate public key with an jwks.json endpoint
+      const jwks = await axios.get(jku);
+      const targetJwk = jwks.data.keys.find(key => key.kid === kid);
+
+      pem = jwkToPem(targetJwk);
+    }
+
+    try {
+      verified = jwt.verify(token, pem, { algorithms: [alg] });
+      console.log('Verified with PEM');
+    } catch (error) {
+      console.error('Invalid Token Error', error.message);
+      return res.status(401).json('Authentication failed');
+    }
+  }
+
+  console.log(`Authenticated Token: ${JSON.stringify(verified)}`);
+
+  return next();
+}
+
+async function authenticateClient(req, res, next) {
+  const { fhirServer, fhirAuthorization } = req.body;
+
+  if (typeof fhirAuthorization === 'undefined') {
+    return next();
+  }
+
+  const tokenObject = {
+    access_token: fhirAuthorization.access_token,
+    expires_in: fhirAuthorization.expires_in,
+    scope: fhirAuthorization.scope,
+  };
+
+  req.fhirClient = new Client({ baseUrl: fhirServer });
+  const { authorizeUrl } = await req.fhirClient.smartAuthMetadata();
+
+  console.log(`${authorizeUrl.protocol}//${authorizeUrl.host}`);
+  console.log(authorizeUrl.pathname);
+
+  // Create a new oAuth2 object using the Client capability statement:
+  const oauth2 = simpleOauthModule.create({
+    client: {
+      id: clientId,
+      secret: clientSecret,
+    },
+    auth: {
+      tokenHost: fhirServer,
+      authorizeHost: `${authorizeUrl.protocol}//${authorizeUrl.host}`,
+      authorizePath: authorizeUrl.pathname,
+    },
+  });
+
+  try {
+    // Create the access token wrapper
+    const { token } = oauth2.accessToken.create(tokenObject);
+
+    console.log('The token is : ', token);
+
+    req.fhirClient.bearerToken = token.access_token;
+  } catch (error) {
+    console.error('Access Token Error', error.message);
+    return res.status(500).json('Authentication failed');
+  }
+
+  return next();
+}
+
+app.get('/cds-services', authenticateEHR, async (req, res) =>
+  res.status(200).json({
+    services: [
+      {
+        enabled: 'true',
+        hook: 'patient-view',
+        id: 'patient-view',
+        name: 'Patient Greeter with Med Count',
+        title: 'Patient Greeter with Med Count',
+        description: 'Example of CDS service greeting patient based on prefetch and counting meds with FHIR Kit client.',
+        prefetch: {
+          patientToGreet: 'Patient/{{context.patientId}}',
+        },
+      },
+    ],
+  }));
+
+app.post('/cds-services/patient-view', [authenticateEHR, authenticateClient], async (req, res) => {
+  let patientGreeting = `Hello ${req.body.prefetch.patientToGreet.resource.name[0].given[0]}! `;
+
+  if (typeof req.fhirClient !== 'undefined') {
+    const medOrders = await req.fhirClient.search({ resourceType: 'MedicationOrder', searchParams: { patient: req.body.patient } });
```

Comment:
I don't see `req.body.patient` as being part of the cds-hooks spec. I think it would be safer to use `request.context.patientId`.